pyvista/plotting/picking.py

```diff
@@ -185,6 +186,9 @@ def visible_pick_call_back(picker, event_id):
                     cids_to_get = tri_smesh.extract_cells(cids)["original_cell_ids"]
                     picked.append(smesh.extract_cells(cids_to_get))
 
+                # memory leak issues on vtk==9.0.20210612.dev0
+                selection.UnRegister(selection)
+
```

Comment:
Another instance of "what?!" (aimed at VTK, not you). Why is `selection` even registered to itself?

Reply:
I can honestly say I tried everything under the sun to figure out the leak here.